src/main/java/year2016/day5/exo/Main.java

```diff
@@ -0,0 +1,140 @@
+package year2016.day5.exo;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hashing;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Main {
+    public static void main(String[] args) {
+        String puzzleInput = "wtnhxymk";
+        //prendre l'input et y ajouter un index croissant
+
+        StringBuilder passwordBuilder = new StringBuilder();
+
+        int index = 0;
+        boolean isPasswordFound = false;
+        while (!isPasswordFound) {
+            String inputToHash = puzzleInput;
+            inputToHash += index;
+
+            passwordBuilder.append(findSixthCharIfHashStartsWithFiveZeros(inputToHash));
+
+            isPasswordFound = passwordBuilder.length() >= 8;
+            index++;
+        }
+
+        System.out.println("answer exo 1 : " + passwordBuilder);
+
+        String passwordForExo2 = findPasswordForExo2(puzzleInput);
+        System.out.println("answer exo 2 : " + passwordForExo2);
+
+    }
+
+    /**
+     * exo 1
+     *
+     * @param input
+     * @return
+     */
+    private static String findSixthCharIfHashStartsWithFiveZeros(String input) {
+        String hash = Hashing.md5().hashString(input, Charsets.UTF_8).toString();
+        //vérifier si le hash commence par 00000
+        Pattern pattern = Pattern.compile("^(00000)");
```

Comment:
Il faut enlever les parenthèses
```suggestion
        Pattern pattern = Pattern.compile("^00000");
```

Utiliser un regex pour récuperer le sixième charactère?

Reply:
parenthèse enlevée.

Une autre fois pour les caractères sur les regex 🥶